### PR TITLE
feat(admin-next, inventory-next, medusa, types): Add admin reservations flow

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add reservations management to admin

### CODE REVIEW

**Comment by olivermrbl** [comment]
@fPolic, @kasperkristensen – could I get one of you to give this one a review? 🙏

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+              name="description"
+              render={({ field }) => {
+                return (
+                  <Form.Item>
+                    <Form.Label optional>{t("fields.description")}</Form.Label>
+                    <Form.Control>
+                      <Textarea
+                        {...field}
+                        placeholder={t(
+                          "inventory.reservation.descriptionPlaceholder"
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx:272]: todo: translation

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+  const { stock_locations } = useStockLocations(
+    {
+      id:
+        selectedInventroyItem?.location_levels?.map(
+          (level: InventoryNext.InventoryLevelDTO) => level.location_id
+        ) ?? [],
+    },
+    {
+      enabled: !!selectedInventroyItem,
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx]: suggestion: either disable location select until the item is selected or show a tooltip for location saying that inventory needs to be selected first

File: packages/admin-next/dashboard/public/locales/en-US/translation.json [hunk 3]
```diff
@@ -338,7 +338,22 @@
...
+      "description": "Description", 
+      "location": "Location",
+      "inStockAtLocation": "In stock at this location",
+      "availableAtLocation": "Available at this location",
+      "reservedAtLocation": "Reserved at this location",
+      "reservedAmount": "Reserve amount",
+      "create": "Create reservation",
+      "itemToReserve": "Item to reserve",
+      "quantityPlaceholder": "How many do you want to reserve?",
+      "descriptionPlaceholde": "What type of reservation is this?"
```
Comment (fPolic) [inline packages/admin-next/dashboard/public/locales/en-US/translation.json]: typo

File: packages/admin-next/dashboard/src/types/api-responses.ts [hunk 4]
```diff
@@ -71,6 +71,13 @@ export type RegionRes = { region: RegionDTO }
 export type RegionListRes = { regions: RegionDTO[] } & ListRes
 export type RegionDeleteRes = DeleteRes
 
+// Regions
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/types/api-responses.ts]: typo

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-detail/components/edit-reservation/components/edit-reservation-form.tsx [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+  const form = useForm<zod.infer<typeof EditReservationSchema>>({
+    defaultValues: getDefaultValues(reservation),
+    resolver: zodResolver(EditReservationSchema),
+  })
+
+  const { mutateAsync } = useUpdateReservationItem(reservation.id)
+
+  const handleSubmit = form.handleSubmit(async (values) => {
+    mutateAsync(values as any, {
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-detail/components/edit-reservation/components/edit-reservation-form.tsx:73]: todo: add toast + suggestion use try catch

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-detail/components/edit-reservation/edit-reservation-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import { useParams } from "react-router-dom"
+import { useReservationItem } from "../../../../../hooks/api/reservations"
+import { useStockLocations } from "../../../../../hooks/api/stock-locations"
+import { useTranslation } from "react-i18next"
+
+export const ReservationEdit = () => {
+  const { id } = useParams()
+  const { t } = useTranslation()
+
+  const { reservation, isLoading, isError, error } = useReservationItem(id!)
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-detail/components/edit-reservation/edit-reservation-modal.tsx]: suggestion: use `isPending` here also, `isLoading` should be enough but `isPending` guarantees data

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-detail/components/reservation-general-section/reservation-general-section.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+export const ReservationGeneralSection = ({
+  reservation,
+}: ReservationGeneralSectionProps) => {
+  const { t } = useTranslation()
+
+  const { inventory_item: inventoryItem, isLoading: isLoadingInventoryItem } =
+    useInventoryItem(reservation.inventory_item_id)
+
+  const { stock_location: location, isLoading: isLoadingLocation } =
+    useStockLocation(reservation.location_id)
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-detail/components/reservation-general-section/reservation-general-section.tsx]: suggeestion: `isPending`

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/components/reservation-list-table/reservation-actions.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      description: t("reservations.deleteWarning"),
+      confirmText: t("actions.delete"),
+      cancelText: t("actions.cancel"),
+    })
+
+    if (!res) {
+      return
+    }
+
+    await mutateAsync()
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/components/reservation-list-table/reservation-actions.tsx:30]: todo: toast

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/components/reservation-list-table/reservation-list-table.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+const PAGE_SIZE = 20
+
+export const ReservationListTable = () => {
+  const { t } = useTranslation()
+
+  const { searchParams } = useReservationTableQuery({
+    pageSize: PAGE_SIZE,
+  })
+  const { reservations, count, isLoading, isError, error } =
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/components/reservation-list-table/reservation-list-table.tsx]: suggestion: isPending

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/components/reservation-list-table/reservation-list-table.tsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+      </div>
+      <DataTable
+        table={table}
+        columns={columns}
+        pageSize={PAGE_SIZE}
+        count={count}
+        isLoading={isLoading}
+        filters={filters}
+        pagination
+        navigateTo={(row) => `${row.id}`}
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/components/reservation-list-table/reservation-list-table.tsx]:
```suggestion
        navigateTo={(row) => row.id}
```

File: packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+    }
+  )
+
+  const { mutateAsync, isPending } = useCreateReservationItem()
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    await mutateAsync(data, {
```
Comment (fPolic) [inline packages/admin-next/dashboard/src/v2-routes/reservations/reservation-list/create-reservation/components/create-reservation-form/create-reservation-form.tsx:100]: todo: toast

**Review by fPolic — COMMENTED** [review]
Looks good, I've added a few minor comments

**Review by fPolic — APPROVED** [review]

**Pull request merged**